src/provider/application.js

```diff
@@ -231,6 +233,12 @@ class Application extends EventEmitter {
   emitError(error) {
     this.emit('xfc.error', error);
   }
+
+  unload() {
+    const nextURL = document.activeElement.href;
+
+    this.trigger('xfc.unload', { url: nextURL });
```

Comment:
NIT: just do
```js
{ url: document.activeElement.href } 
```

Also I don't think you can assume there is always an active element and it has an href. Is the url needed for anything? Else I would just suggest removing it

Reply:
The URL and target are from the redirect dictionary. They are both used for the redirect dictionary to indicate where the next page will be so we can anticipate/match the path and the target is used to indicate if we are opening in a new tab or not. 

If we remove them from here we need to keep the redirect dictionary SDK event intact.

Reply:
Instead of using the checkRedirect from sdk, in theory we could use this unload trigger. The good thing about the unload trigger is that I believe it will only activate if the target is _self by default. However, the URL could still be useful.

Reply:
You believe it does or it does? 😛

Reply:
Hmm... looking into it, there are a few different types of targets I didn't know exist, such as _parent and _top. I'm not sure what the use cases would be for those target types, but I think the primary reason to use the target is for links which open up in a new tab (which would not activate the unload trigger). However, maybe there is a reason to know more of the fine detail of which type of target is selected in a link?

Reply:
We just use target=_blank for no